### DISCUSSION TITLE
coordinate reference system (CRS)

### DISCUSSION THREAD

**Discussion by dahlalex** [opengeos/leafmap#215] [Q&A] [answered]
What are the CRS:s that are supported by leafmap? Is there a way to convert the CRS?

**Comment by giswqs** [comment, marked as answer]
For vector, leafmap supports any CRS supported by GeoPandas. Under the hood, all CRS are converted to WGS 84 (lat/lon) to be displayed on the map. 

For raster, leafmap supports any CRS supported by ipyleaflet, which only supports a limited number of CRS. See https://github.com/jupyter-widgets/ipyleaflet/blob/stable/ipyleaflet/projections.py

**Comment by dahlalex** [comment]
Thanks, So in order for data to be displayed correctly on the map, it should be converted first to WGS84 EPSG:4326. Can one change the CRS of the basemap instead?

**Reply by giswqs** [reply]
You don't have to convert data to EPSG:4326. Leafmap will do that automatically for you. 

You can change the basemap projection if you want, but only a limited number of CRS are supported. 
https://github.com/jupyter-widgets/ipyleaflet/blob/stable/ipyleaflet/projections.py

**Reply by rh-afk** [reply]
can we use custom projection as mentioned in this link 
[https://github.com/jupyter-widgets/ipyleaflet/blob/master/examples/CustomProjections.ipynb]
and use leafmap to display the map?

**Reply by giswqs** [reply]
Yes. Leafmap supports anything ipyleaflet supports.

**Reply by rh-afk** [reply]
Thanks @giswqs for your information, unfortunately I just reproduce example from above link with my local server tile layer , in ipyleaflet it displayed tilelayer but when use the leafmap, it shows nothing. Is there any consideration that I must aware about that in leafmap.

Thanks

**Reply by giswqs** [reply]
Works fine on my end. 

```python
import leafmap
from ipyleaflet import (
    basemaps,
    projections,
)

m = leafmap.Map(
    center=(90, 0),
    zoom=0,
    basemap=basemaps.NASAGIBS.BlueMarble3413,
    crs=projections.EPSG3413.NASAGIBS,
)
m
```

![image](https://github.com/opengeos/leafmap/assets/5016453/2fd54b1c-0a75-4e0b-a32f-970d458152fd)

**Reply by rh-afk** [reply]
Thanks @giswqs , if I want to use arc gis tile server is it the same code too? and how about plot some data point from dataframe or add additional layer to that map?
kindly please advise.

thanks
